src/ai/ai.cpp

```diff
@@ -81,69 +107,66 @@ void form_alliances(sys::state& state) {
 
+static inline bool ai_is_close_enough(sys::state& state, dcon::nation_id target, dcon::nation_id from) {
+	auto target_continent = state.world.province_get_continent(state.world.nation_get_capital(target));
+	auto source_continent = state.world.province_get_continent(state.world.nation_get_capital(from));
+	return (target_continent == source_continent) || bool(state.world.get_nation_adjacency_by_nation_adjacency_pair(target, from));
+}
+
 bool ai_will_accept_alliance(sys::state& state, dcon::nation_id target, dcon::nation_id from) {
 	if(!state.world.nation_get_ai_is_threatened(target))
 		return false;
 
 	if(state.world.nation_get_ai_rival(target) == from || state.world.nation_get_ai_rival(from) == target)
 		return false;
-
-	auto target_continent = state.world.province_get_continent(state.world.nation_get_capital(target));
-	auto source_continent = state.world.province_get_continent(state.world.nation_get_capital(from));
-
-	bool close_enough = (target_continent == source_continent) || bool(state.world.get_nation_adjacency_by_nation_adjacency_pair(target, from));
-
-	if(!close_enough)
+	
+	// Same rival equates to instantaneous alliance (we benefit from more allies against a common enemy)
+	if(state.world.nation_get_ai_rival(target) == state.world.nation_get_ai_rival(from))
+		return true;
+	
+	// Otherwise we may consider alliances only iff they are close to our continent or we are adjacent
+	if(!ai_is_close_enough(state, target, from))
 		return false;
-
-	auto target_score = state.world.nation_get_military_score(target);
-	auto source_score = state.world.nation_get_military_score(from);
-
-	return source_score * 2 >= target_score;
+	
+	// And also if they're powerful enough to be considered for an alliance
+	auto target_score = estimate_strength(state, target);
+	auto source_score = estimate_strength(state, from);
+	return source_score * 2.5f >= target_score;
 }
 
 void explain_ai_alliance_reasons(sys::state& state, dcon::nation_id target, text::layout_base& contents, int32_t indent) {
 
 	text::add_line_with_condition(state, contents, "ai_alliance_1", state.world.nation_get_ai_is_threatened(target), indent);
 
-
-	auto target_continent = state.world.province_get_continent(state.world.nation_get_capital(target));
-	auto source_continent = state.world.province_get_continent(state.world.nation_get_capital(state.local_player_nation));
-
-	bool close_enough = (target_continent == source_continent) || bool(state.world.get_nation_adjacency_by_nation_adjacency_pair(target, state.local_player_nation));
-
-	text::add_line_with_condition(state, contents, "ai_alliance_2", close_enough, indent);
+	text::add_line_with_condition(state, contents, "ai_alliance_2", ai_is_close_enough(state, target, state.local_player_nation), indent);
 
 	text::add_line_with_condition(state, contents, "ai_alliance_3", state.world.nation_get_ai_rival(target) != state.local_player_nation && state.world.nation_get_ai_rival(state.local_player_nation) != target, indent);
 
```

Comment:
Since you added an additional condition for when the ai will accept an alliance, you need to add the condition in this section and create text explaining it.